From cd4fd70f3651c8efb8437c1519ce7cbe6670ddb4 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 25 Feb 2013 10:56:21 +0800
Subject: ENGR00251630 MX6DQ: do not power off ARM in suspend on TO1.1

System will resume back failed which caused by IPG clock glich issue. The issue
(TKT094231) has been fixed on MX6DQ TO1.2 and MX6DL TO1.1.It is hard to
reproduced on MX6DQ TO1.1/1.0 with 0xffffffff in GPC_PGC_CPU_PUPSCR
which implemented on GA release. But it is easy to reproduced with 0xf0f0 on
mainline now. As we focus on latest formal chip, the problem is cleaned up until
now. We will aign with the workaround on MX6DL which do not power off ARM in
suspend.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 arch/arm/mach-mx6/pm.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/pm.c b/arch/arm/mach-mx6/pm.c
index 998051d..9f736e4 100644
--- a/arch/arm/mach-mx6/pm.c
+++ b/arch/arm/mach-mx6/pm.c
@@ -1,5 +1,5 @@
 /*
- *  Copyright (C) 2011-2012 Freescale Semiconductor, Inc. All Rights Reserved.
+ *  Copyright (C) 2011-2013 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -312,9 +312,13 @@ static int mx6_suspend_enter(suspend_state_t state)
 	}
 	mx6_suspend_store();
 
-	/* i.MX6dl TO1.0 TKT094231: can't support ARM_POWER_OFF mode */
+	/*
+	 * i.MX6dl TO1.0/i.MX6dq TO1.1/1.0 TKT094231: can't support
+	 * ARM_POWER_OFF mode.
+	 */
 	if (state == PM_SUSPEND_MEM &&
-		(mx6dl_revision() == IMX_CHIP_REVISION_1_0)) {
+		((mx6dl_revision() == IMX_CHIP_REVISION_1_0) || mx6q_revision()
+		<= IMX_CHIP_REVISION_1_1)) {
 		state = PM_SUSPEND_STANDBY;
 	}
 
-- 
1.7.9.5

